src/azure-cli-core/azure/cli/core/_profile.py

```
@@ -385,16 +385,13 @@ def logout_all(self):
         identity.logout_all_users()
         identity.logout_all_service_principal()
 
-    def get_login_credentials(self, resource=None, subscription_id=None, aux_subscriptions=None, aux_tenants=None):
+    def get_login_credentials(self, subscription_id=None, aux_subscriptions=None, aux_tenants=None):
```

For Track 1 SDK, the `resource` of the access token is managed by Azure CLI, _not_ Track 1 SDK. `resource` is kept as a property of the credential returned by `get_login_credentials()`. When SDK client calls `signed_session()` to acquire the access token, CLI provides the access token for `resource`.

```
                                returns
get_login_credentials(resource) ------>  credential[resource]
                                             ^
                                             |   calls signed_session()
                                             |
                                  Track 1 SDK client()
```

For Track 2 SDK, the `scopes`(`resource/.default`) of the access token is managed by SDK client instead. The `scopes` is passed to Track 2 SDK via `credential_scopes` argument when creating the client instance. The SDK client keeps it and passes it to `get_token()` when acquiring the access token from the credential.

```
                            returns
get_login_credentials()  ------------->  credential()
                                             ^
                                             |   calls get_token(scopes)
                                             |
                             Track 2 SDK client(credential_scopes)
```

Since Track 1 SDK is no longer supported, Azure CLI doesn't need to manage `resource` anymore, so `resource` argument is dropped.

---

`resource` was marked as Track 1 only in 2021 (#19853):

https://github.com/Azure/azure-cli/blob/967983c749c70e5e7dec5f1f84dae9032dc1008d/src/azure-cli-core/azure/cli/core/_profile.py#L391

---

`resource` was marked as Track 1 only in 2021 (#19853):

https://github.com/Azure/azure-cli/blob/967983c749c70e5e7dec5f1f84dae9032dc1008d/src/azure-cli-core/azure/cli/core/_profile.py#L391

If a command module or extension still passes `resource` to `get_login_credentials()`, it is very likely `credential_scopes` is not set correctly. This will cause failure in other clouds other than the public cloud, as the token scope doesn't match what is declared in these clouds.

Removing `resource` argument makes it fail early instead of failing in deeper or more complex code paths.